tests/impl/async_get_result.cpp

```diff
@@ -161,28 +181,30 @@ TEST_F(async_get_result, should_post_callback_with_error_if_consume_input_failed
     Sequence s;
 
     // Post self to strand
+    EXPECT_CALL(m.strand, on_work_started()).InSequence(s).WillOnce(Return());
     EXPECT_CALL(m.executor, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
     EXPECT_CALL(m.strand, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
-    EXPECT_CALL(m.callback, context_preserved()).InSequence(s).WillOnce(Return());
 
     // Wait for read while is_busy() returns true
     EXPECT_CALL(m.connection, is_busy()).InSequence(s).WillOnce(Return(true));
     EXPECT_CALL(m.socket, async_read_some(_)).InSequence(s).WillOnce(InvokeArgument<0>(error_code{}));
-    EXPECT_CALL(m.strand, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
-    EXPECT_CALL(m.callback, context_preserved()).InSequence(s).WillOnce(Return());
+    EXPECT_CALL(m.strand, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
 
     // Consume input
     EXPECT_CALL(m.connection, consume_input()).InSequence(s).WillOnce(Return(0));
 
     // Cancel all io
-    EXPECT_CALL(m.socket, cancel(_)).WillOnce(Return());
+    EXPECT_CALL(m.socket, cancel(_)).InSequence(s).WillOnce(Return());
 
-    // Post callback with error
+    // Post callback with
+    EXPECT_CALL(m.callback, get_executor()).WillOnce(Return(ozo::tests::executor {&m.callback_executor}));
+    EXPECT_CALL(m.callback_executor, on_work_started()).InSequence(s).WillOnce(Return());
     EXPECT_CALL(m.executor, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
-    EXPECT_CALL(m.strand, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
-    EXPECT_CALL(m.callback, context_preserved()).InSequence(s).WillOnce(Return());
+    EXPECT_CALL(m.callback_executor, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
     EXPECT_CALL(m.callback, call(error_code{ozo::error::pg_consume_input_failed}, _))
         .InSequence(s).WillOnce(Return());
+    EXPECT_CALL(m.callback_executor, on_work_finished()).InSequence(s).WillOnce(Return());
+    EXPECT_CALL(m.strand, on_work_finished()).InSequence(s).WillOnce(Return());
```

Comment:
on_work_started()/on_work_finished() are really important for the tests? You can add for them default behaviour in fixture to do not saturate tests with not important things.

Reply:
Done